api/v2/openapi.yaml

```diff
@@ -496,6 +557,111 @@ components:
           maxItems: 100
           items:
             $ref: '#/components/schemas/Location'
+    Badge:
+      title: Badge
+      description: Badges for certain actions
+      type: string
+      nullable: true
+      enum:
+        - next_level
+        - next_level_class
+        - report
+        - new_report
+        - two_shops
+        - shop_first_report
+        - week_1
+        - week_3
+        - week_full
+        - week_special
+        - trusted_shop
+        - trusted_user
+    NewBadge:
+      title: NewBadge
+      description: latest Badges of a specific user
+      type: object
+      nullable: false
+      required:
+        - badges
+        - reason
+      properties:
+        badges:
+          schema:
+            type: array
+            items:
+              $ref: '#/components/schemas/Badge'
+        reason:
+          type: string
+          example: occupancy
+    AllBadges:
+      title: AllBadges
+      description: all Badges of a specific user plus the total ammount of badges
+      type: object
+      nullable: false
+      required:
+        - badgeAndCounter
+        - countAllBadges
+      properties:
+        BadgeAndCounter:
+          schema:
+            type: array
+            items:
+              $ref: '#/components/schemas/BadgeAndCounter'
+        NumberOfBadges:
+          type: integer
+          example: 5
+    BadgeAndCounter:
+      title: BadgeAndCounter
+      description: shows how often a user earnd the same badge
+      type: object
+      nullable: false
+      required:
+        - badge
+        - counter
+      properties:
+        badge:
+          type: 
+            $ref: '#/components/schemas/Badge'        
+        counter:
+          type: integer
+          example: 1
+    LevelAndXp:
+      title: LevelAndXp
+      description: level and xp of one user
+      type: object
+      nullable: false
+      required:
+        - level
+        - stage
+        - xpTotal
+        - xpNeeded
+        - xpCurrent
+      properties:
+        stage:
+          type: 
+            $ref: '#/components/schemas/Stage'        
+        level:
+          type: integer
+          example: 5
+        xpTotal:
+          type: integer
+          example: 150
+        xpNeeded:
+          type: integer
+          example: 25
+        xpCurrent:
+          type: integer
+          example: 10
```

Comment:
Das würde ich selbst nochmal in ein Objekt fassen, weil ja alles nur mit `xp` zu tun hat, also:
```json
{
"stage": "novice",
"level": 1,
"xp": {
    "total": 150,
    "needed": 25,
    "current": 10
}}
```

bzw:
```yaml
properties:
  stage:
    $ref: '#/components/schemas/Stage'
  level:
    type: integer
  xp:
    type: object
    properties:
      total:
        type: integer
      needed:
        type: integer
      current:
        type: integer
```

Reply:
Sieht so wirklich besser aus. Werd ich übernehmen.